packages/ruby/lib/readme/filter.rb

```diff
@@ -0,0 +1,43 @@
+class Filter
+  def self.for(filter_params = nil, allow_only = nil)
+    if !filter_params.nil? && !allow_only.nil?
+      raise "Can only supply either filter_params or allow_only"
+    end
+
+    unless filter_params.nil?
+      return FilterParams.new(filter_params)
+    end
+
+    unless allow_only.nil?
+      return AllowOnly.new(allow_only)
+    end
+
+    None.new
```

Comment:
Thoughts on making a single 4-way conditional here instead of relying on early returns?